sdk/data/azcosmos/cosmos_error.go

```diff
@@ -11,26 +11,18 @@ import (
 	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
-// cosmosError is used as base error for any error response from the Cosmos service.
-type cosmosError struct {
-	body        string
-	code        string
-	rawResponse *http.Response
-}
+// CosmosError is used as base error for any error response from the Cosmos service.
+type CosmosError struct {
```

Comment:
Should we get rid of this and just use azcore.ResponeError?

Reply:
I don't mind using azcore.ResponseError, is that a new interface/type in the new azcore version?

Reply:
Yeah it's something we added in `v0.21.0`.  I've updated the PR with this change.